docs/prometheus.md

```diff
@@ -9,245 +9,358 @@ sidebar_label: Prometheus
+- `openebs-device` - Using this option, it will create Kubernetes Local PVs using the block devices attached to the node. Select this option when you want to dedicate a complete block device on a node to a Prometheus application pod and other device for Alert manager. You can customize which devices will be discovered and managed by OpenEBS using the instructions [here](/docs/next/uglocalpv-device.html#optional-block-device-tagging). 
 
-4. **Create Storage Class** : 
+The Storage Class `openebs-device` has been chosen to deploy Prometheus Operator in the Kubernetes cluster.
 
-   StorageClass is the interface through which most of the OpenEBS storage policies  are defined. See Prometheus Storage Class section below. 
+**Note:** Ensure that you have two disks with the required capacity is added to the corresponding nodes prior to Prometheus installation. In this example, we have added two 100G disks to each node.
 
-5. **Configure PVC** : Prometheus needs only one volume to store the data. See PVC example spec below.
+### Installing Prometheus Operator
+
+In this section, we will install the Prometheus operator. We will later deploy the latest available version of Prometheus application. The following are the high-level overview.
+
+- Label the nodes
+- Fetch and update Prometheus repository
+- Configure Prometheus Helm `values.yaml`
+- Create namespace for installing application
+- Install Prometheus operator
+
+#### Label the nodes
+
+Label the nodes with custom label so that Prometheus application will be deployed only on the matched Nodes. Label each node with `node=prometheus`. We have used this label in the Node Affinity for Prometheus and Alert Manager instances. This will ensure to schedule the Prometheus and Alert Manager pods to deploy only on the labelled nodes.
+
+In this example, we used the following command to label our nodes.
+
+```
+kubectl label nodes ip-192-168-30-203.ap-south-1.compute.internal node=prometheus
+kubectl label nodes ip-192-168-55-47.ap-south-1.compute.internal node=prometheus
+kubectl label nodes ip-192-168-68-28.ap-south-1.compute.internal node=prometheus
+```
+
+#### Fetch and update Prometheus Helm repository
+
+```
+$ helm repo add prometheus-community https://prometheus-community.github.io/helm-charts
+$ helm repo update
+```
 
-6. **Launch and test Prometheus**:
+#### Configure Prometheus Helm `values.yaml`
 
-   Run `kubectl apply -f <prometheus.yaml>` to see Prometheus running. For more information on configuring more services to be monitored, see Prometheus documentation.
+Download `values.yaml` which will modify before installing Prometheus using Helm.
 
+```none
+wget https://raw.githubusercontent.com/prometheus-community/helm-charts/main/charts/kube-prometheus-stack/values.yaml
+```
+
+Perform the following changes:
+
+- Update `fullnameOverride: "new" `
+
+- Update `prometheus.prometheusSpec.serviceMonitorSelectorNilUsesHelmValues` as `false`
+
+- Update `prometheus.prometheusSpec.replicas` as `3`
```

Comment:
Can you emphasize why this is required? Is this because we are letting Prometheus keep a copy of the metrics on different nodes?

Reply:
Since we are using OpenEBS LocalPV device, the application should require the maintain the copies considering the node down scenarios. Since app is STS, we are going with our usual recommended node count as 3.

Reply:
Also, using Local PV - there may be a limitation that if Prometheus pod is down, then data from that time period when the pod was done will not be available. This may be ok, if you are using Prometheus mainly for real time monitoring and alerts. 

However, if you want to use Prometheus for saving historical data, you will need to consider using cstor or other durable engines. 

Can I request you to push the old cStor content that is in this document to https://github.com/openebs/cstor-operators/tree/master/docs. Create a new directory called "workloads" and then show the instructions to run prometheus operator on top of cstor csi (with cspc class). 

When moving into the directory, we don't need to show all other details like setting up cstor. we can directly jump into using the prometheus operator with `cstor-csi` storage class. This instruction to setup `cstor-csi` are available in: https://github.com/openebs/cstor-operators/blob/master/docs/quick.md

Reply:
@kmova I have updated Prometheus operator deployment using Local PV device on OpenEBS docs and Prometheus operator deployment using cStor engine at https://github.com/openebs/cstor-operators/blob/master/docs/workload/prometheus-operator/prometheus.md. Please check...